File: Python/Problems/Problems/problem3731.py
import heapq
import math
import queue
import sys

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter, deque

from .problem import Problem

class Problem3731(Problem):
    def test(self) -> bool:
        nums = [1, 4, 2, 5]

        expected = [3]

        result = self.findMissingElements(nums)

        print1DArray(result)

        return expected == result

    def findMissingElements(self, nums: List[int]) -> List[int]:
        mx = max(nums)
        mn = min(nums)
        l = mx - mn + 1
        present = [False] * l
        for num in nums:
            present[num - mn] = True

        result = []
        for i in range(l):
            if not present[i]:
                result.append(mn + i)
        return result

File: C++/Problems/Problems/problem_3731.cpp
#include "problem_3731.h"

bool problem_3731::test() {
    std::vector nums { 1, 4, 2, 5 };

    std::vector expected { 3 };

    auto result = findMissingElements(nums);

    print1DVector(result);

    return expected == result;
}

std::vector<int> problem_3731::findMissingElements(std::vector<int> &nums) {
    int mx = nums[0];
    int mn = nums[0];
    for (auto num : nums)
    {
        mx = std::max(mx, num);
        mn = std::min(mn, num);
    }
    int l = mx - mn + 1;
    std::vector<bool> present(l);
    for (auto num : nums)
        present[num - mn] = true;
    std::vector<int> result;
    for (int i = 0; i < l; ++i)
        if (!present[i])
            result.push_back(mn + i);
    return result;
}
